src/nspawn/nspawn-mount.c

```diff
@@ -691,6 +691,10 @@ static int parse_mount_bind_options(const char *options, unsigned long *mount_fl
                         flags |= MS_REC;
                 else if (streq(word, "norbind"))
                         flags &= ~MS_REC;
+                else if (streq(word, "idmap"))
+                        *idmapped = true;
+                else if (streq(word, "noidmap"))
+                        *idmapped = false;
```

Comment:
i'd prefer if we'd not touch these values unless the function fully succeeded, like for mount_flags and mount_opts. i.e. make changes in local variable and after everything succeded propagate it into the parameters